src/main/java/com/powsybl/openloadflow/network/LfAction.java

```diff
@@ -138,11 +169,27 @@ public static Optional<LfAction> create(Action action, LfNetwork lfNetwork, Netw
             case HvdcAction.NAME:
                 return create((HvdcAction) action, lfNetwork);
 
+            case ShuntCompensatorPositionAction.NAME:
+                return create((ShuntCompensatorPositionAction) action, lfNetwork);
+
             default:
                 throw new UnsupportedOperationException("Unsupported action type: " + action.getType());
         }
     }
 
+    private static Optional<LfAction> create(ShuntCompensatorPositionAction action, LfNetwork lfNetwork) {
+        LfShunt shunt = lfNetwork.getShuntById(action.getShuntCompensatorId());
+        if (shunt instanceof LfShuntImpl) { // no svc here
+            if (shunt.isVoltageControlEnabled()) {
+                throw new UnsupportedOperationException("Shunt compensator position action: voltage controller shunt not supported");
```

Comment:
I think just adding a test for this exception will make coverage > 90%

Reply:
I do or you do?